lib/src/messages/balance_transaction.dart

```diff
@@ -5,20 +5,50 @@ enum _BalanceTransactionObject { balance_transaction }
 
 /// https://stripe.com/docs/api/balance_transactions/object
 @JsonSerializable()
-class BalanceTransaction {
+class BalanceTransaction extends Message {
   final _BalanceTransactionObject object;
   final String id;
   final int amount;
+  final int fee;
+  final List<FeeDetails> feeDetails;
   final int net;
   final String currency;
+
   BalanceTransaction({
     required this.object,
     required this.id,
     required this.amount,
+    required this.fee,
+    required this.feeDetails,
     required this.net,
     required this.currency,
   });
+
   factory BalanceTransaction.fromJson(Map<String, dynamic> json) =>
       _$BalanceTransactionFromJson(json);
+
+  @override
   Map<String, dynamic> toJson() => _$BalanceTransactionToJson(this);
 }
+
+@JsonSerializable()
+class FeeDetails {
```

Comment:
```suggestion
class FeeDetails extends Message {
```

Reply:
Currently, the main point of introducing this `Message` parent class was to allow the Resource class to specify its contents (`class Resource<T extends Message>`). Ultimately, I decided not to put the Message class on things like `FeeDetails`, `PaymentMethodDetailsCard`, `SubscriptionItem`, `DataList`, etc because these are nested objects that (currently) _can't_ be requested directly from the API and therefore it shouldn't be possible to create a (for example) `class FeeDetailsResource extends Resource<FeeDetails>`.

TBH, I went back and forth on this decision. Let me know your thoughts and I'd be happy to change.